### PR TITLE
fix issue #71 -Wrong contribution type being added to .all-contributorsrc 

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
**What**:
Issue https://github.com/jfmengels/all-contributors-cli/issues/71

**Why**:
Wrong contribution type being added to .all-contributorsrc despite the error

**How**:
- I have included that change in `prompt.js`
- I am checking if its a valid type from `utils`
- Currently i am passing the filtered types further down the program . eg
 if user types `add foo code,bar,video`
 - code and video are added 
and if user types all wrong types then nothing will be added with an error (please tell me if this is what you are looking for)?

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged 
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
Yes I can add

**Comment by M-ZubairAhmed** [comment]
should I make a new file under test folder for prompt, since that's where I have inserted the code

**Comment by kentcdodds** [comment]
Either way

**Comment by M-ZubairAhmed** [comment]
this also checks off one from https://github.com/jfmengels/all-contributors-cli/issues/6
> Contribution type not found.

File: src/contributors/__tests__/prompt.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  })
+})
+
+test(`should throw error if no contribution types were included in add command`, () => {
+  const options = fixtures()
+  const username = 'userName'
+  const contributions = ''
+  expect(() => prompt(options, username, contributions)).toThrow(
+    `No contribution type found in the input. Did you forget to include them in the add command?`,
+  )
```
Comment (machour) [inline src/contributors/__tests__/prompt.js]: If no contribution types are found in the add command, the user should be prompted for them.

I should be able to just run the cli without any argument, be prompted for the operation I want to do (add/generate), then be prompted for the rest of the arguments.

Currently your patch breaks that:

screen shot 2017-12-25 at 12 20 13 pm

File: README.md [hunk 2]
```diff
@@ -35,9 +35,10 @@ specification.
 
 - [Installation](#installation)
 - [Usage](#usage)
+  - [Generating the contributors list](#generating-the-contributors-list)
+  - [Add/update contributors](#addupdate-contributors)
+  - [Check for missing contributors](#check-for-missing-contributors)
 - [Configuration](#configuration)
-- [Inspiration](#inspiration)
-- [Other Solutions](#other-solutions)
```
Comment (machour) [inline README.md]: This should be dealt with in another PR.

File: README.md [hunk 3]
```diff
@@ -167,13 +168,11 @@ Thanks goes to these wonderful people
 ([emoji key](https://github.com/kentcdodds/all-contributors#emoji-key)):
 
 <!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
-
```
Comment (machour) [inline README.md]: Did you remove that space by hand, or did the command output change? 🤔

File: src/contributors/__tests__/prompt.js (hunk 1 above)
Comment (M-ZubairAhmed) [reply src/contributors/__tests__/prompt.js]: ok let us handle this in another pr then

**Comment by M-ZubairAhmed** [comment]
i have reverted all the change to unwanted files

**Comment by M-ZubairAhmed** [comment]
currently the toc is being generated when i commit a readme file automatically

**Comment by M-ZubairAhmed** [comment]
I have added my self as contributor in this https://github.com/jfmengels/all-contributors-cli/pull/79 pr

**Comment by machour** [comment]
Thank you for the rollbacks :)

Tested the new PR, I still see it breaking things:

```bash
🍺  ~/all-contributors-cli (master)*$ ./src/cli.js
? What do you want to do? Add a new contributor or add a new contribution type
Invalid contribution type/s entered
```

Normally here, I would have been asked first for the login, then for the contributions types (with a nice selection interface). Your PR seems to force the user to give all arguments on the command line. (did you know about this?)

**Comment by M-ZubairAhmed** [comment]
I did not know about this as in the issue it was mentioned error has occured because of 
> ~/git-point (master)*$ yarn contributors:add foo bar

So i tested the code with that command and did not look at the CLI. What would you like me to do?

**Comment by M-ZubairAhmed** [comment]
can you take a look at this now? Cli seems to be working. May be i am missing some edge cases of the usage

**Comment by M-ZubairAhmed** [comment]
any status on this :-)

**Comment by machour** [comment]
@M-ZubairAhmed busy week for me, I'll try to recheck this soon.

**Comment by M-ZubairAhmed** [comment]
hi,
sorry could check on this earlier, @machour  could you please check this

**Comment by M-ZubairAhmed** [comment]
Hi,
Sorry any updates on this?

File: src/contributors/prompt.js [hunk 4]
```diff
@@ -69,10 +69,28 @@ function getQuestions(options, username, contributions) {
...
+function getValidUserContributions(options, contributions) {
+  const validContributionTypes = util.contributionTypes(options)
+  const userContributions = contributions && contributions.split(',')
+
+  const validUserContributions = _.filter(
+    userContribution => validContributionTypes[userContribution] !== undefined,
+  )(userContributions)
+
+  if (_.isEmpty(validUserContributions)) {
+    throw new Error('Invalid contribution type(s) entered')
```
Comment (kentcdodds) [inline src/contributors/prompt.js]: Could we make the error include a comma-separated list of the invalid contribution types?

Comment (M-ZubairAhmed) [reply src/contributors/prompt.js]: you mean to say, have a list of **valid** contribution types?

Comment (kentcdodds) [reply src/contributors/prompt.js]: I want them to be able to see which contribution type they entered which was invalid so they know not to enter that again.

Comment (M-ZubairAhmed) [reply src/contributors/prompt.js]: added please take a look :+1:
